pkg/k8s/watchers/watcher.go

```diff
@@ -535,7 +534,11 @@ func (k *K8sWatcher) enableK8sWatchers(ctx context.Context, resourceNames []stri
 
 func (k *K8sWatcher) k8sServiceHandler() {
 	eventHandler := func(event k8s.ServiceEvent) {
-		defer event.SWG.Done()
+		startTime := time.Now()
+		defer func() {
+			event.SWG.Done()
+			k.K8sServiceEventProcessed(event.Action.String(), event.Service.String(), startTime)
```

Comment:
Not sure this warrants a separate call. Perhaps we could update the metric in line here? 

nit: Don't think we need to define a var, we could justl call defer func() {...}(time.Now())

Reply:
The defer func will be evaluated after the method finish it's execution so it will no longer meter the method execution duration if we remove the var.

Reply:
I think this is right, you can leave the defer of event.SWG.Done() and have a separate inline defer, because the used values are bound immediately and won't be affected by later execution.

Reply:
Thanks for the suggestion, removed the var declaration and added it as argument for the defer function.